pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -185,14 +222,43 @@ func (c *podIpMetricsCollector) setupDirectoryWatcher(dir string) error {
 
 		for {
 			select {
-			case _, ok := <-watcher.Events:
+			case e, ok := <-watcher.Events:
 				if !ok {
 					glog.Error("watcher is not ok")
 					return
 				}
 				if err := c.listIpAddresses(dir); err != nil {
 					return
 				}
+				names := strings.Split(e.Name, "/")
+				f := names[len(names)-1]
+				reusedIp, ok := reusedMap[f]
```

Comment:
It is better to create a helper function for all of the logic that follows. A good indicator of when you need a helper function is when the line becomes beyond 80 characters. This particular logic is heavily indented making it less readable.

Reply:
Done.